File: business/src/main/java/work/beltran/discogsbrowser/business/SearchInteractor.java
package work.beltran.discogsbrowser.business;

import rx.Observable;
import work.beltran.discogsbrowser.api.model.record.Record;

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
public interface SearchInteractor {
    Observable<Record> search(final String query, final int nextPage);
}

File: data-repository/src/main/java/work/beltran/discogsbrowser/business/SearchInteractor.kt
package work.beltran.discogsbrowser.business

import io.reactivex.Single
import work.beltran.discogsbrowser.api.model.record.RecordApi

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
interface SearchInteractor {
    fun search(query: String): Single<List<RecordApi>>
}
